backend-shared/apis/managed-gitops/v1alpha1/gitopsdeploymentmanagedenvironment_types.go

```diff
@@ -24,14 +24,54 @@ const (
 	ManagedEnvironmentStatusConnectionInitializationSucceeded = "ConnectionInitializationSucceeded"
 )
 
-// GitOpsDeploymentManagedEnvironmentSpec defines the desired state of GitOpsDeploymentManagedEnvironment
+// The GitOpsDeploymentManagedEnvironment CR describes a remote cluster which the GitOps Service will deploy to, via Argo CD.
+// This resource references a Secret resource, of type managed-gitops.redhat.com/managed-environment, that contains the cluster credentials.
+// The Secret should contain credentials to a ServiceAccount/User account on the target cluster.
+// This is referred to as the Argo CD 'ServiceAccount' below.
 type GitOpsDeploymentManagedEnvironmentSpec struct {
-	APIURL                     string   `json:"apiURL"`
-	ClusterCredentialsSecret   string   `json:"credentialsSecret"`
-	AllowInsecureSkipTLSVerify bool     `json:"allowInsecureSkipTLSVerify"`
-	CreateNewServiceAccount    bool     `json:"createNewServiceAccount,omitempty"`
-	Namespaces                 []string `json:"namespaces,omitempty"`
-	ClusterResources           bool     `json:"clusterResources,omitempty"`
+
+	// APIURL is the URL of the cluster to connect to
+	APIURL string `json:"apiURL"`
+
+	// ClusterCredentialsSecret is a reference to a Secret that contains cluster connection details. The cluster details should be in the form of a kubeconfig file.
+	ClusterCredentialsSecret string `json:"credentialsSecret"`
+
+	// AllowInsecureSkipTLSVerify controls whether Argo CD will accept a Kubernetes API URL with untrusted-TLS certificate.
+	// Optional: If true, the GitOps Service will allow Argo CD to connect to the specified cluster even if it is using an invalid or self-signed TLS certificate.
+	// Defaults to false.
+	AllowInsecureSkipTLSVerify bool `json:"allowInsecureSkipTLSVerify"`
+
+	// CreateNewServiceAccount controls whether Argo CD will use the ServiceAccount provided by the user in the Secret, or if a new ServiceAccount
+	// should be created.
+	//
+	// Optional, default to false.
+	//
+	// - If true, the GitOps Service will automatically create a ServiceAccount/ClusterRole/ClusterRoleBinding on the target cluster,
+	//   using the credentials provided by the user in the secret.
+	//   - Argo CD will then be configured to deploy with that new ServiceAccount.
+	//
+	// - Default: If false, it is assumed that the credentials provided by the user in the Secret are for a ServiceAccount on the cluster, and
+	//   Argo CD will be configred to use the ServiceAccount referenced by the Secret of the user. No new ServiceAccount will be created.
+	//   - This should be used, for example, when the ServiceAccount Argo CD is using will not full cluster access (*/*/* at cluster scope)
```

Comment:
`...will not full cluster access...` should be `...will not have full cluster access...`.  It might read better to say `...does not have full cluster access...`